### PR TITLE
introduce type failures in dispatchers

### PR DESCRIPTION
**Pull request by pmeier**
As discussed offline yesterday, we should have a proper type check in our dispatchers. We had a concern about JIT making this more complicated, but this doesn't seem to be the case. With the changes in this PR, the following works fine:

```pycon
>>> torch.jit.script(F.erase)
>>> torch.jit.script(F.resize)
```

In CI we additionally test that calling the scripted kernel works and this is green as well: https://github.com/pytorch/vision/actions/runs/3572617624/jobs/6005708359

cc @vfdev-5 @datumbox @bjuncek

### CODE REVIEW

**Comment by datumbox** [comment]
@pmeier I'm surprised that JIT works. Probably it evaluates that the first if is always true during scripting and thus clips the entire other code. This is great, LGTM. Ping me when you introduce it to all dispatchers to stamp.

**Comment by vfdev-5** [comment]
yes, the graph behind resize is the following:
```
graph(%inpt.1 : Tensor,
      %size.1 : int[],
      %interpolation.1 : Enum<__torch__.torchvision.transforms.functional.InterpolationMode>,
      %max_size.1 : int?,
      %antialias.1 : bool?):
  %26 : Function = prim::Constant[name="resize_image_tensor"]()
  %inpt.7 : Tensor = prim::unchecked_cast(%inpt.1)
  %inpt.15 : Tensor = prim::unchecked_cast(%inpt.7)
  %27 : Tensor = prim::CallFunction(%26, %inpt.15, %size.1, %interpolation.1, %max_size.1, %antialias.1) # /var/folders/8r/t4lgs4t95xjgyj6g63kz10mm0000gn/T/ipykernel_93605/4050849040.py:34:15
  return (%27)
```

File: torchvision/prototype/transforms/functional/_meta.py [hunk 1]
```diff
@@ -103,7 +113,7 @@ def get_spatial_size(inpt: features.InputTypeJIT) -> List[int]:
         return get_spatial_size_image_tensor(inpt)
     elif isinstance(inpt, (features.Image, features.Video, features.BoundingBox, features.Mask)):
         return list(inpt.spatial_size)
-    else:
+    elif isinstance(inpt, PIL.Image.Image):
```
Comment (datumbox) [inline torchvision/prototype/transforms/functional/_meta.py:116]: We need an else statement with the `TypeError`.

**Review by datumbox — APPROVED** [review]
LGTM, just one comment that needs to be addressed. Also can you double confirm that no dispatcher is missed (it's hard to check methods that are not modified in the PR)?

**Review by datumbox — APPROVED** [review]

**Pull request merged**